src/biogeophys/SnowHydrologyMod.F90

```diff
@@ -100,20 +101,6 @@ module SnowHydrologyMod
   integer, parameter, public :: LoTmpDnsSlater2017            = 2    ! For temperature below -15C use equation from Slater 2017
   integer, parameter, public :: LoTmpDnsTruncatedAnderson1976 = 1    ! Truncate low temp. snow density from the Anderson-1976 version at -15C
 
-  ! Definition of snow pack vertical structure
-  ! Hardcoded maximum of 12 snowlayers, this is checked elsewhere (controlMod.F90)
-  ! The bottom layer has no limit on thickness, hence the last element of the dzmax_*
-  ! arrays is 'huge'.
-  real(r8), parameter :: dzmin(12) = &       ! minimum of top snow layer
-               (/ 0.010_r8, 0.015_r8, 0.025_r8, 0.055_r8, 0.115_r8, 0.235_r8, &
-                  0.475_r8, 0.955_r8, 1.915_r8, 3.835_r8, 7.675_r8, 15.355_r8 /)
-  real(r8), parameter :: dzmax_l(12) = &     ! maximum thickness of layer when no layers beneath
-               (/ 0.03_r8, 0.07_r8, 0.18_r8, 0.41_r8, 0.88_r8, 1.83_r8, &
-                  3.74_r8, 7.57_r8, 15.24_r8, 30.59_r8, 61.3_r8, huge(1._r8)  /)
-  real(r8), parameter :: dzmax_u(12) = &     ! maximum thickness of layer when layers beneath
-               (/ 0.02_r8, 0.05_r8, 0.11_r8, 0.23_r8, 0.47_r8, 0.95_r8, &
-                  1.91_r8, 3.83_r8, 7.67_r8, 15.35_r8, 30.71_r8, huge(1._r8)  /)
```

Comment:
In reference to [this comment](https://github.com/ESCOMP/ctsm/pull/792/files#r317752274), the original code assigned huge(1._r8) to dzmax_l(12) and dzmax_u(12). So to match the original behavior, I think I need to do the following:

`if (nlevsno >= 12) then`
`dzmax_l = huge(1._r8)`
`dzmax_u = huge(1._r8)`
`end if`

Is this what you were suggesting @billsacks ?

Reply:
More precisely...
`if (j >= 12) then`
`dzmax_l(j) = ...`
`dzmax_u(j) = ...`

Reply:
Changing mind:
I prefer the first if statement with these corrections:
`dzmax_l(nlevsno) = ...`
`dzmax_u(nlevsno) = ...`

This replicates the current behavior for nlevsno = 12. If some day we allow nlevsno > 12, this just sets the bottom layer to huge.

Reply:
Yes, the last comment is what I was thinking (where the ... should be `huge(1._r8)`).

Reply:
i.e., the bottom layer should be set to huge, regardless of the number of snow layers. I now realize that that isn't exactly what the original code did, but I think that was the intent.

Reply:
I guess it would be good to verify that that does the right thing. Either (a) make that change and verify that it doesn't change answers for, say, a 5-layer case (a CLM45 case), or (b) don't make that change, and ensure that having dzmax_l and dzmax_u NOT set to huge for nlevsno gives the same answers as setting it to huge for a 12-layer case. I prefer (a), but don't feel strongly about it.

Reply:
I also prefer (a).

Reply:
> I guess it would be good to verify that that does the right thing. Either (a) make that change and verify that it doesn't change answers for, say, a 5-layer case (a CLM45 case), or [...]

The hobart test-suite includes a few clm45 cases and they all PASS.